samples/net/openthread/coap/src/coap_utils.c

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) 2024 Alexandre Bailon
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(coap);
+
+#include <coap_utils.h>
+
+static uint8_t g_coap_buf[COAP_MAX_BUF_SIZE];
+static uint8_t g_coap_dev_id[COAP_DEVICE_ID_SIZE];
+
+#ifdef CONFIG_OT_COAP_SAMPLE_SERVER
+static void coap_default_handler(void *context, otMessage *message,
+				 const otMessageInfo *message_info)
+{
+	ARG_UNUSED(context);
+	ARG_UNUSED(message);
+	ARG_UNUSED(message_info);
+
+	LOG_INF("Received CoAP message that does not match any request "
+		"or resource");
+}
+#endif
+
+static int coap_req_send(const char *addr, const char *uri, uint8_t *buf, int len,
+			 otCoapResponseHandler handler, void *ctx, otCoapCode code)
+{
+	otInstance *ot;
+	otMessage *msg;
+	otMessageInfo msg_info;
+	otError err;
+	int ret;
+
+	ot = openthread_get_default_instance();
+	if (!ot) {
+		LOG_ERR("Failed to get an OpenThread instance");
+		return -ENODEV;
+	}
+
+	memset(&msg_info, 0, sizeof(msg_info));
+	otIp6AddressFromString(addr, &msg_info.mPeerAddr);
+	msg_info.mPeerPort = OT_DEFAULT_COAP_PORT;
+
+	msg = otCoapNewMessage(ot, NULL);
+	if (!msg) {
+		LOG_ERR("Failed to allocate a new CoAP message");
+		return -ENOMEM;
+	}
+
+	otCoapMessageInit(msg, OT_COAP_TYPE_CONFIRMABLE, code);
+
+	err = otCoapMessageAppendUriPathOptions(msg, uri);
+	if (err != OT_ERROR_NONE) {
+		const char *err_str = otThreadErrorToString(err);
```

Comment:
Maybe instead of defining `err_str` variable in error handling code, just use `otThreadErrorToString(err)` directly in the log function? This applies to other functions too.